include/zephyr/drivers/psi5/psi5.h

```diff
@@ -0,0 +1,247 @@
+{
+	const struct psi5_driver_api *api = (const struct psi5_driver_api *)dev->api;
+
+	if (api->start_sync) {
+		return api->start_sync(dev, channel);
+	}
+
+	return -ENOSYS;
+}
+
+/**
+ * @brief Stop the sync pulse generator on a specific channel
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param channel The hardware channel of the driver instance.
+ * @retval 0 if successful.
+ * @retval -EINVAL if an invalid channel is given.
+ * @retval -EALREADY if the device is already stopped.
+ * @retval -EIO General input/output error, failed to stop device.
+ */
+__syscall int psi5_stop_sync(const struct device *dev, uint8_t channel);
+
+static inline int z_impl_psi5_stop_sync(const struct device *dev, uint8_t channel)
+{
+	const struct psi5_driver_api *api = (const struct psi5_driver_api *)dev->api;
+
+	if (api->stop_sync) {
+		return api->stop_sync(dev, channel);
+	}
+
+	return -ENOSYS;
+}
+
+/**
+ * @brief Transmitting PSI5 data on a specific channel
+ *
+ * @param dev       Pointer to the device structure for the driver instance.
+ * @param channel   The hardware channel of the driver instance.
+ * @param data      PSI5 data to transmit.
+ * @param timeout   Timeout waiting for ready to transmit new data.
+ * @param callback  Optional callback for when the frame was sent or a
+ *                  transmission error occurred. If ``NULL``, this function is
+ *                  blocking until frame is sent.
+ * @param user_data User data to pass to callback function.
+ *
+ * @retval 0 if successful.
+ * @retval -EINVAL if an invalid channel is given.
+ * @retval -ENOTSUP if an unsupported parameter was passed to the function.
+ * @retval -ENETDOWN if PSI5 is in stopped state.
+ * @retval -EIO if a general transmit error occurred.
+ * @retval -EAGAIN on timeout.
+ */
+__syscall int psi5_send(const struct device *dev, uint8_t channel, const uint64_t data,
+			k_timeout_t timeout, psi5_tx_callback_t callback, void *user_data);
+
+static inline int z_impl_psi5_send(const struct device *dev, uint8_t channel, const uint64_t data,
+				   k_timeout_t timeout, psi5_tx_callback_t callback,
+				   void *user_data)
+{
+	const struct psi5_driver_api *api = (const struct psi5_driver_api *)dev->api;
+
+	if (api->send) {
+		return api->send(dev, channel, data, timeout, callback, user_data);
+	}
+
+	return -ENOSYS;
+}
+
+/**
+ * @brief Add a callback function to handle messages received for a specific channel
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param channel The hardware channel of the driver instance.
+ * @param callback_configs The callback configurations.
+ * @retval 0 if successful.
+ * @retval -EINVAL if an invalid channel is given.
+ */
+__syscall int psi5_register_callback(const struct device *dev, uint8_t channel,
+				     struct psi5_rx_callback_configs callback_configs);
+
+static inline int z_impl_psi5_register_callback(const struct device *dev, uint8_t channel,
```

Comment:
Any pre-condition before this API is invoked? Pls mention in API prototype.

Reply:
I mentioned  the pre-condition